plugin/src/main/java/org/gradle/testretry/internal/executer/RetryTestResultProcessor.java

```diff
@@ -108,7 +111,12 @@ public void completed(Object testId, TestCompleteEvent testCompleteEvent) {
 
                 boolean failedInPreviousRound = previousRoundFailedTests.remove(className, name);
                 if (failedInPreviousRound && testCompleteEvent.getResultType() == SKIPPED) {
```

Comment:
I suggest collapsing these two conditions into a single boolean property to better convey the intent. Maybe something like this:

```
boolean shouldRetrySkippedTestThatPreviouslyFailed = failedInPreviousRound && testCompleteEvent.getResultType() == SKIPPED && failOnSkippedAfterRetry;
if (shouldRetrySkippedTestThatPreviouslyFailed) {
   addRetry(descriptor);
}
```